server/lib/manifest.py

```diff
@@ -0,0 +1,389 @@
+                "@id": "https://www.materialsdatafacility.org/",
+                "@type": "Organization",
+                "legalName": "Materials Data Facility",
+                "Description": "A simple way to publish, discover, and access materials datasets"
+            }
+    }
+
+    def create_basic_attributes(self, tale):
+        """
+        Returns a portion of basic attributes in the manifest
+        :param tale: A Tale that is being described
+        :return: Basic information about the Tale
+        """
+
+        return {
+            "@id": 'https://data.wholetale.org/api/v1/tale/' + str(tale['_id']),
+            "createdOn": str(tale['created']),
+            "schema:name": tale['title'],
+            "schema:description": tale.get('description', str()),
+            "schema:category": tale['category'],
+            "schema:identifier": str(tale['_id']),
+            "schema:version": tale['format'],
+            "schema:image": tale['illustration'],
+            "aggregates": list(),
+            "Datasets": list()
+        }
+
+    def add_tale_creator(self, tale, user):
+        """
+        Adds basic information about the Tale author
+        :param tale: The Tale being described
+        :param user: The user whose information is being used
+        """
+
+        tale_user = self.userModel.load(tale['creatorId'], user=user)
+        self.manifest['createdBy'] = {
+            "@id": tale['authors'],
+            "@type": "schema:Person",
+            "schema:givenName": tale_user.get('firstName', ''),
+            "schema:familyName": tale_user.get('lastName', ''),
+            "schema:email": tale_user.get('email', '')
+        }
+
+    def create_context(self):
+        """
+        Creates the manifest namespace. When a new vocabulary is used, it shoud
+        get added here.
+        :return: A structure defining the used vocabularies
+        """
+        return {
+            "@context": [
+                "https://w3id.org/bundle/context",
+                {"schema": "http://schema.org/"}
+            ]
+        }
+
+    def create_dataset_record(self, user, folder_id):
+        """
+        Creates a record that describes a Dataset
+        :param user: The user
+        :param folder_id: Folder that represents a dataset
+        :return: Dictionary that describes a dataset
+        """
+        folder = self.folderModel.load(folder_id,
+                                       user=user,
+                                       force=True)
+        if folder:
+            meta = folder.get('meta')
+            if meta:
+                provider = meta.get('provider')
+                if provider:
+                    return {
+                        "@id": meta.get('identifier'),
+                        "@type": "Dataset",
+                        "name": folder['name'],
+                        "identifier": meta.get('identifier'),
+                        "publisher": self.publishers[provider]}
```

Comment:
IMHO it would be slightly cleaner if you could:
1. load Folder with `exc=True`
2. explicitly do `meta = folder['meta'] ; provider = meta['provider']`
3. `try/except` everything catching `(KeyError, ValidationException)`
This would allow you to get rid of all the *if* statements.

Reply:
Done in 651f7a4